File: F. We Were Both Children.py
tests = int(float(input("")))
for i in range(tests):
    length = int(input(""))
    array = tuple(map(int, input().split()))
    d = {}
    for i in range(length):
        
        jump = array[i]
        if jump>length:
            continue
        print("jump:",jump)
        pos = 0
        for j in range(length):
            print(pos,end=" ")
            d[pos] = d.get(pos,0) + 1
            pos += jump
        print()
    if 0 in d:
        d.pop(0)
    print("d:",d)
    print("ans:")
    if len(d)==0:
        print(0)
    else:
        print(max(d.values()))

File: F_We_Were_Both_Children.cpp
#include <iostream>
#include <vector>
using namespace std;

int solve(vector<int>& arr, int n) {
    vector<int> ans(n + 1, 0);
    int Max = 0;

    for (int i : arr) {
        for (int j = i; j <= n; j += i) {
            ans[j]++;
            Max = max(Max, ans[j]);
        }
    }

    return Max;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);

        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }

        cout << solve(arr, n) << endl;
    }

    return 0;
}
